core/src/window_service.rs

```diff
@@ -24,6 +24,9 @@ fn retransmit_blobs(blobs: &[SharedBlob], retransmit: &BlobSender, id: &Pubkey)
     for blob in blobs {
         // Don't add blobs generated by this node to the retransmit queue
         if blob.read().unwrap().id() != *id {
+            let mut w_blob = blob.write().unwrap();
```

Comment:
I had to move this here from packet.rs because it was causing gossip sig verify to fail when I called set_forwarded(false) on the blob, eventhough forwarded was already false. :\

Reply:
I don't get it.  Moar background?

Reply:
So I don't really understand why, but Gossip's signature verification begins failing when I make this a generic change such that we're always updating the blob as it comes off of the wire (in packet::recv_blob). Even when the value was already false, set_forwarded(false) would cause the finally deserialized blob to fail gossip verification. I suspect gossip data to_blobs is not respecting data boundaries?